packages/core-data/src/entities.ts

```diff
@@ -12,157 +12,296 @@ import { __ } from '@wordpress/i18n';
 		baseURL: '/wp/v2/taxonomies',
 		baseURLParams: { context: 'edit' },
 		plural: 'taxonomies',
 		label: __( 'Taxonomy' ),
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'sidebar',
 		kind: 'root',
 		baseURL: '/wp/v2/sidebars',
 		plural: 'sidebars',
 		transientEdits: { blocks: true },
 		label: __( 'Widget areas' ),
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'widget',
 		kind: 'root',
 		baseURL: '/wp/v2/widgets',
 		baseURLParams: { context: 'edit' },
 		plural: 'widgets',
 		transientEdits: { blocks: true },
 		label: __( 'Widgets' ),
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'widgetType',
 		kind: 'root',
 		baseURL: '/wp/v2/widget-types',
 		baseURLParams: { context: 'edit' },
 		plural: 'widgetTypes',
 		label: __( 'Widget types' ),
-	},
-	{
+	} as const ),
+	defineEntity( {
 		label: __( 'User' ),
 		name: 'user',
 		kind: 'root',
 		baseURL: '/wp/v2/users',
 		baseURLParams: { context: 'edit' },
 		plural: 'users',
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'comment',
 		kind: 'root',
 		baseURL: '/wp/v2/comments',
 		baseURLParams: { context: 'edit' },
 		plural: 'comments',
 		label: __( 'Comment' ),
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'menu',
 		kind: 'root',
 		baseURL: '/wp/v2/menus',
 		baseURLParams: { context: 'edit' },
 		plural: 'menus',
 		label: __( 'Menu' ),
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'menuItem',
 		kind: 'root',
 		baseURL: '/wp/v2/menu-items',
 		baseURLParams: { context: 'edit' },
 		plural: 'menuItems',
 		label: __( 'Menu Item' ),
 		rawAttributes: [ 'title', 'content' ],
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'menuLocation',
 		kind: 'root',
 		baseURL: '/wp/v2/menu-locations',
 		baseURLParams: { context: 'edit' },
 		plural: 'menuLocations',
 		label: __( 'Menu Location' ),
 		key: 'name',
-	},
-	{
+	} as const ),
+	defineEntity( {
 		name: 'navigationArea',
 		kind: 'root',
 		baseURL: '/wp/v2/block-navigation-areas',
 		baseURLParams: { context: 'edit' },
 		plural: 'navigationAreas',
 		label: __( 'Navigation Area' ),
 		key: 'name',
-		getTitle: ( record ) => record?.description,
-	},
-	{
+		getTitle: ( record: EntityTypes.NavigationArea< 'edit' > | null ) =>
+			record?.description,
+	} as const ),
+	defineEntity( {
 		label: __( 'Global Styles' ),
 		name: 'globalStyles',
 		kind: 'root',
 		baseURL: '/wp/v2/global-styles',
 		baseURLParams: { context: 'edit' },
 		plural: 'globalStylesVariations', // should be different than name
 		getTitle: ( record ) => record?.title?.rendered || record?.title,
-	},
-	{
+	} as const ),
+	defineEntity( {
 		label: __( 'Themes' ),
 		name: 'theme',
 		kind: 'root',
 		baseURL: '/wp/v2/themes',
 		baseURLParams: { context: 'edit' },
 		key: 'stylesheet',
-	},
-	{
+	} as const ),
+	defineEntity( {
 		label: __( 'Plugins' ),
 		name: 'plugin',
 		kind: 'root',
 		baseURL: '/wp/v2/plugins',
 		baseURLParams: { context: 'edit' },
 		key: 'plugin',
-	},
-];
+	} as const ),
+] as const;
```

Comment:
do we need the runtime function to wrap the types?
did you try this?

```ts
export const defaultEntities: EntityDefinition[] = [
	{ … },
	…
];
```

Reply:
The type annotation didn't work, but I found a different workaround [on StackOverflow](https://stackoverflow.com/questions/57069802/as-const-is-ignored-when-there-is-a-type-definition). Tl;dr, it is possible to wrap the entire config in a function that guarantees type safety and extracts the type definition:

```ts
export const defaultEntities = entitiesConfig( [
	{
		label: __( 'Base' ),
		kind: 'root',
		name: '__unstableBase',
		baseURL: '/',
	},
```